### PR TITLE
Use GraphQL AsyncClientSession with automatic retries

### PR DESCRIPTION
**Pull request by jschlyter**
## Summary by CodeRabbit

- **New Features**
	- Introduced enhanced GraphQL session management with retry logic for improved error handling.
	- Added a new asynchronous function for establishing GraphQL sessions.
	- Implemented new constants for connection and execution retries to optimize performance.
- **Bug Fixes**
	- Improved error handling for GraphQL operations by raising exceptions when sessions are uninitialized.

### CODE REVIEW

**Comment by jschlyter** [comment]
Perhaps @BJReplay can take a look?

**Comment by BJReplay** [comment]
> Perhaps @BJReplay can take a look?

Code runs in Dev Container applying your changes to latest (1.10.0) release manually.  I can't really test on a live car, so I'm only really seeing this return `Unknown` for a lot of entities, but it runs.

I note a lot of ruff warnings against existing code, so won't point those out - no doubt you can spot those as easily as I can.  I will note one against new lines in this PR.

File: custom_components/polestar_api/pypolestar/polestar.py [hunk 1]
```diff
@@ -276,34 +282,33 @@ async def _query_graph_ql(
...
+                exc.errors
+                and exc.errors[0].get("extensions", {}).get("code") == "UNAUTHENTICATED"
+            ):
+                self.latest_call_code = 401
+                raise PolestarNotAuthorizedException(exc.errors[0]["message"]) from exc
+            self.latest_call_code = 500
+            raise PolestarApiException from exc
+        except Exception as exc:
+            self.logger.debug("GraphQL Exception: %s", str(exc))
+            raise exc
```
Comment (BJReplay) [inline custom_components/polestar_api/pypolestar/polestar.py:311]: This is showing Use `raise` without specifying exception name Ruff [TRY201](https://docs.astral.sh/ruff/rules/verbose-raise) in my Dev Container test environment, suggesting you can use use `raise` to throw the exception.

**Review by BJReplay — APPROVED** [review]
Looks good by my slowly developing python skilz that are interpreting what extensions like ruff tell me to do.

File: custom_components/polestar_api/pypolestar/polestar.py (hunk 1 above)
Comment (jschlyter) [reply custom_components/polestar_api/pypolestar/polestar.py:311]: Correct, you can raise without `exc` - not sure I agree it is better though

**Pull request merged**